src/main/java/mate.academy/lib/Injector.java

```diff
@@ -8,6 +20,48 @@
     }
 
     public Object getInstance(Class<?> interfaceClazz) {
-        return null;
+        Class<?> clazz = findImplementation(interfaceClazz);
+
+        if (!clazz.isAnnotationPresent(Component.class)) {
+            throw new RuntimeException("Class " + interfaceClazz.getName()
+                    + " must be annotated with @Component");
+        }
+
+        Field[] declaredField = clazz.getDeclaredFields();
+        Object clazzImplementationInstance = createNewInstance(clazz);
+        for (Field field : declaredField) {
+            if (field.isAnnotationPresent(Inject.class)) {
+                Object fieldInstance = createNewInstance(findImplementation(field.getType()));
+                try {
+                    field.setAccessible(true);
+                    field.set(clazzImplementationInstance, fieldInstance);
+                } catch (IllegalAccessException e) {
+                    throw new RuntimeException("Cannot initialize field value");
+                }
+            }
+        }
+
+        return clazzImplementationInstance;
+    }
+
+    private Object createNewInstance(Class<?> clazz) {
+        try {
+            Constructor<?> constructor = clazz.getConstructor();
+            return constructor.newInstance();
+        } catch (NoSuchMethodException | InvocationTargetException | InstantiationException
+                 | IllegalAccessException e) {
+            throw new RuntimeException("Cannot create a new instance of: " + clazz.getName());
+        }
+    }
+
+    private Class<?> findImplementation(Class<?> interfaceClazz) {
+        Map<Class<?>, Class<?>> interfaceImplementations = new HashMap<>();
```

Comment:
The map for interface implementations should be a class field and initialized using `Map.of()`. Initializing it within the `findImplementation` method is not ideal for maintainability and clarity.

Reply:
fix comment